components/TemporaryNotification.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Box } from './Grid';
+import { FormattedMessage } from 'react-intl';
+import styled from 'styled-components';
+import { position } from 'styled-system';
+import StyledButton from './StyledButton';
+
+const Notification = styled.div`
+  top: 0;
+  box-sizing: border-box;
+  width: 100%;
+  padding: 12px 24px;
+  background: #1869f5;
+  font-size: 14px;
+  line-height: 21px;
+  color: white;
+  display: flex;
+  align-items: center;
+  z-index: 99;
+
+  ${position}
+`;
+
+const DismissButton = styled(StyledButton).attrs({
+  buttonStyle: 'secondary',
+  buttonSize: 'tiny',
+  color: 'blue.100',
+  borderColor: 'blue.100',
+})`
+  background: transparent;
+  border: 1px solid white;
+  font-weight: normal;
+
+  &:hover:not(:active):not(:disabled) {
+    background: rgba(245, 250, 255, 0.25);
+  }
+`;
+
+/**
+ * Displays a temporary notification that can be dismissed by the user.
+ */
+const TemporaryNotification = ({ children, position, onDismiss }) => {
+  const [isDismissed, setDismissed] = React.useState(false);
+
+  if (isDismissed) {
+    return null;
+  }
+
+  return (
+    <Notification position={position}>
+      <Box mr="auto" />
+      <Box>{children}</Box>
+      <Box ml="auto">
+        <DismissButton
+          ml={2}
+          onClick={() => {
+            setDismissed(true);
+            if (onDismiss) {
+              onDismiss();
+            }
+          }}
```

Comment:
🍨 
Should we call `onDismiss` before actually hiding the banner?
How much are we allowing this logic to be extended?
Perhaps something like this could be more generic and compatible with different use cases:
```suggestion
          onClick={async () => {
            if (onDismiss) {
              await onDismiss();
            }
            setDismissed(true);
          }}
```

Reply:
The one I have in mind for this would be an async call (like `DismissibleMessage`) that saves some info when the notification is dismissed. In such case, I think it's better to close the notification right away when clicked, no need to wait fo the async call to finish.

If we call `onDismiss` first, there's a change that parents will adapt the page and eventually remove the notification from the DOM, resulting in `Calling setState on a unmoounted component` warnings when trying to `setDismissed(true)`.

That being said no problem to change it if there's a use case, but we need to also add a loading state for the button to not make it look unresponsive.

Reply:
Merging to get this live. Still open for discussion, feel free to refactor it if you need to!